src/skin/legacyskinparser.cpp

```diff
@@ -304,19 +306,40 @@ QWidget* LegacySkinParser::parseSkin(QString skinPath, QWidget* pParent) {
 
     SkinManifest manifest = getSkinManifest(skinDocument);
 
-    // Apply SkinManifest attributes.
+    // Apply SkinManifest attributes by looping through the proto.
     for (int i = 0; i < manifest.attribute_size(); ++i) {
         const SkinManifest::Attribute& attribute = manifest.attribute(i);
         if (!attribute.has_config_key()) {
             continue;
         }
-        ConfigKey configKey = ConfigKey::parseCommaSeparated(
-            QString::fromStdString(attribute.config_key()));
 
         bool ok = false;
         double value = QString::fromStdString(attribute.value()).toDouble(&ok);
         if (ok) {
-            ControlObject::set(configKey, value);
+            ConfigKey configKey = ConfigKey::parseCommaSeparated(
+                    QString::fromStdString(attribute.config_key()));
+            // Set the specified attribute, possibly creating the control
+            // object in the process.
+            bool created = false;
+            // If there is no existing value for this CO in the skin,
+            // update the config with the specified value. If the attribute
+            // is set to persist, the value will be read when the control is created.
+            if (m_pConfig->getValueString(configKey).isEmpty()) {
```

Comment:
This is a hack that relies on how the rest of the code works. IMHO It fixes a general issue in the CO setup code, but at the wrong place. We have similar issues when introducing a new  persistent CO with a non 0.0 default value in the rest of the code. Mixxx can't do this by now, so please fix that.

The root issue is located in ControlDoublePrivate::initialize()
around this line 
https://github.com/mixxxdj/mixxx/blob/c631ab52641bcdd7fec0c6caba54f7d5d2c66d0b/src/control/control.cpp#L49